### PR TITLE
enable eigen tensor module

### PR DESCRIPTION
**Pull request by JensWehner**
This serves to replace boost multiarray. Eigen::Tensor is now used by Tensorflow, i.e. supported by Google. I think the support is good enough now to warrant deprecating boost::multiarray.

### CODE REVIEW

File: include/votca/tools/eigen.h [hunk 1]
```diff
@@ -44,6 +44,7 @@
 #pragma GCC diagnostic ignored "-Wint-in-bool-context"
 #endif
 #include <Eigen/Eigen>
+#include <unsupported/Eigen/CXX11/Tensor>
```
Comment (junghans) [inline include/votca/tools/eigen.h]: Is it really in a folder `unsupported/` ?

Comment (JensWehner) [reply include/votca/tools/eigen.h]: yeah. It is an unofficial module of eigen but since google uses it in tensorflow. I think it is good enough

Comment (junghans) [reply include/votca/tools/eigen.h]: ok

Comment (JensWehner) [reply include/votca/tools/eigen.h]: https://stackoverflow.com/questions/41518379/why-was-eigen-chosen-for-tensorflow

**Pull request merged**